### PR TITLE
Flakes: empty vtdataroot before starting a new vreplication e2e test

### PR DESCRIPTION
**Pull request by rohit-nayak-ps**
## Description

Multiple replication tests have started failing because the disk space if filling up. This PR cleans up the vtdataroot for all vreplication e2e tests while setting up the test cluster.

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Review by GuptaManan100 — APPROVED** [review]

File: go/test/endtoend/vreplication/cluster_test.go [hunk 1]
```diff
@@ -323,6 +323,9 @@ func init() {
 func NewVitessCluster(t *testing.T, name string, cellNames []string, clusterConfig *ClusterConfig) *VitessCluster {
 	vc := &VitessCluster{Name: name, Cells: make(map[string]*Cell), ClusterConfig: clusterConfig}
 	require.NotNil(t, vc)
+
+	vc.CleanupDataroot(t)
```
Comment (mattlord) [inline go/test/endtoend/vreplication/cluster_test.go:327]: Don't we need to do this in teardown? I thought each workflow (each test run) used its own unique VTDATAROOT, so this new root will already be empty, no?

File: go/test/endtoend/vreplication/cluster_test.go [hunk 2]
```diff
@@ -358,6 +361,24 @@ func NewVitessCluster(t *testing.T, name string, cellNames []string, clusterConf
...
+	ci, ok := os.LookupEnv("CI")
+	if !ok || strings.ToLower(ci) != "true" {
+		// Leave the directory in place to support local debugging.
+		return
+	}
+	dir := vc.ClusterConfig.vtdataroot
+	log.Infof("Deleting vtdataroot %s", dir)
+	err := os.RemoveAll(dir)
+	require.NoError(t, err)
+	err = os.Mkdir(dir, 0755)
```
Comment (mattlord) [inline go/test/endtoend/vreplication/cluster_test.go:378]: We don't need `other` to have any privileges do we? i.e. I think we should use 750 unless there's some special reason.

File: go/test/endtoend/vreplication/cluster_test.go (hunk 1 above)
Comment (rohit-nayak-ps) [reply go/test/endtoend/vreplication/cluster_test.go]: Right, I added to teardown. I also wanted to make sure that the vtdataroot was actually empty: while we should always start with a new random directory, was wondering if during the failures we somehow get an existing vtdataroot. Will look for that the next time I see a "disk space not available". But it shouldn't hurt to keep it I guess.

File: go/test/endtoend/vreplication/cluster_test.go (hunk 2 above)
Comment (rohit-nayak-ps) [reply go/test/endtoend/vreplication/cluster_test.go]: used `0700`

**Review by mattlord — APPROVED** [review]
Thank you! ❤️
